github/compliance/model.py

```diff
@@ -80,14 +84,74 @@ class ScanState(enum.Enum):
 
 @dataclasses.dataclass(kw_only=True)
 class ScanResult:
+    # sentinel class to detect if latest processing date is missing
+    @dataclasses.dataclass(frozen=True)
+    class NoLatestProcessingDate:
+        pass
+
     scanned_element: Target
+    discovery_date: datetime.date = datetime.date.today()
+    latest_processing_date: datetime.date | NoLatestProcessingDate | None = None
 
     state: ScanState = ScanState.SUCCEEDED
 
+    _no_latest_processing_date_sentinel: NoLatestProcessingDate = NoLatestProcessingDate()
+
     @property
     def scan_succeeded(self) -> bool:
         return self.state in [ScanState.SUCCEEDED, ScanState.SKIPPED]
 
+    @property
+    def severity(self) -> Severity:
+        # has to be defined by enclosing class
+        return self._no_latest_processing_date_sentinel
+
+    @property
+    def has_latest_processing_date(self) -> bool:
+        if self.latest_processing_date is self._no_latest_processing_date_sentinel:
+            return False
+        return True
+
+    def calculate_latest_processing_date(
+        self,
+        max_processing_days: MaxProcessingTimesDays=None,
+        delivery_svc_client: delivery.client.DeliveryServiceClient=None,
+        repository: github3.repos.Repository=None,
+    ):
+        if self.severity is None: # explicitly check for `None` in case severity is `0`
+            self.latest_processing_date = None
+            return
+        if self.severity is self._no_latest_processing_date_sentinel:
+            self.latest_processing_date = self._no_latest_processing_date_sentinel
+            return
+
+        if not max_processing_days:
+            max_processing_days = MaxProcessingTimesDays()
+        max_days = max_processing_days.for_severity(severity=self.severity)
+        date = self.discovery_date + datetime.timedelta(days=max_days)
+
+        # if the max processing time is 0 days, keep `latest_processing_date` to
+        # make sure, the finding is assigned to the current sprint during issue
+        # creation instead of a sprint lying in the past
+        if delivery_svc_client and max_days > 0:
+            try:
+                import github.compliance.report as gcr
+                target_sprint = gcr._target_sprint(
+                    delivery_svc_client=delivery_svc_client,
+                    latest_processing_date=date,
+                )
+                target_milestone = gcr._target_milestone(
+                    repo=repository,
+                    sprint=target_sprint,
+                )
+                date = target_milestone.due_on.date()
+            except Exception as e:
+                import traceback
+                traceback.print_exc()
+                print(f'{e=}')
```

Comment:
nit: will this print add any additional output that traceback.print_exc() did not already print?